### DISCUSSION TITLE
Mocking fs module for a 3rd party library

### DISCUSSION THREAD

**Discussion by PaulRosset** [vitest-dev/vitest#5589] [Q&A] [answered]
Hello,

I'm currently trying to migrate from Jest to Vitest.
However, I'm encountering an issue when I want to mock the `fs` module with `memfs` for an external library that I'm using.

With Jest, the following was working fine:
`jest.mock('node:fs', () => memFs)`
`jest.mock('fs', () => memFs)`

But using vitest, it makes the external library that is performing a call to `fs` not mocked, however, calls that are performed outside of this lib are working fine.
But for some reason, I cant mock any call to `fs` that have been called inside this external library.

```
| my-app <-- My app is performing some `fs` calls that are mocked right with `memfs`
| node_modules
 | external-lib <-- This external lib is performing some `fs` calls that I want to be mocked, but for some reason, they are not...
```

I also did a reproduction repo here:
- https://stackblitz.com/edit/stackblitz-starters-dewcrr (Vitest - **KO**)
- https://stackblitz.com/edit/stackblitz-starters-ps1fn9 (Jest - **OK**)

Does anyone know what I could be missing here?

**Comment by hi-ogawa** [comment, marked as answer]
`vi.mock` only intercepts `import` inside the files which are processed by Vite. In your reproduction, the external dependency `to-vfile`'s `import` will be kep as is, so it acceess non-mocked `fs` modules.

One way to extend this to external dependency is to use `server.deps.inline: ["to-vfile"]` option, which will make Vite process `to-vfile` just like your source code. Here is an updated reproduction:
https://stackblitz.com/edit/stackblitz-starters-rwafjx?file=vitest.config.ts

**Reply by PaulRosset** [reply]
Superb! Thank you a lot for your help @hi-ogawa :) 

Do you think it could make sense to add it to the documentation on the ["Migrating from jest"](https://vitest.dev/guide/migration.html#migrating-from-jest) part ?

I can do it if you think it could make sense.

**Reply by hi-ogawa** [reply]
I'm not the only one to decide (and also I'm not so familiar with what jest does in comparison), but yeah, it sounds nice to mention something in the doc. Feel free to try!

**Reply by shirakaba** [reply]
This example saved me, thanks! I don't understand why `__mocks__/fs.cjs` wasn't used automatically for my subdependency and this approach successfully worked around it. I had to use your per-test `import './_mocks/fs.js'` approach instead.

To be clear, `__mocks__/fs.cjs` was applied for usages of `fs` directly within my project, but not for inlined subdependencies.

Unfortunately my wrists are hurting too much right now to prepare a Stackblitz repro, sorry! 😅

**Comment by TenPotatoes** [comment]
Type definitions indicate it should be `test.deps.inline`, not `server.deps.inline`

I had an import like `"my-package/util"`, and what worked was

```
test: {
  deps: {
     inline: [ "my-package" ]
  }
}
```
